libs/azure-storage/tests/unit_tests/test_document_loaders.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, Callable, Iterator, Tuple
+from unittest.mock import MagicMock, patch
+
+import azure.identity
+import pytest
+from azure.storage.blob import BlobClient, ContainerClient
+from azure.storage.blob._download import StorageStreamDownloader
+
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+
+@pytest.fixture
+def account_url() -> str:
+    return "https://testaccount.blob.core.windows.net"
+
+
+@pytest.fixture
+def container_name() -> str:
+    return "test-container"
+
+
+@pytest.fixture
+def blobs() -> list[dict[str, str]]:
+    return [
+        {"blob_name": "text_file.txt", "blob_content": "test content"},
+        {"blob_name": "json_file.json", "blob_content": "{'test': 'test content'}"},
+        {"blob_name": "csv_file.csv", "blob_content": "col1,col2\nval1,val2"},
+    ]
+
+
+@pytest.fixture
+def create_azure_blob_storage_loader(
+    account_url: str, container_name: str
+) -> Callable[..., AzureBlobStorageLoader]:
+    def _create_azure_blob_storage_loader(**kwargs: Any) -> AzureBlobStorageLoader:
+        return AzureBlobStorageLoader(
+            account_url,
+            container_name,
+            **kwargs,
+        )
+
+    return _create_azure_blob_storage_loader
+
+
+@pytest.fixture
+def mock_blob_clients(
+    account_url: str, container_name: str, blobs: list[dict[str, str]]
+) -> Callable[[str], MagicMock]:
+    def _get_blob_client(blob_name: str) -> MagicMock:
```

Comment:
So instead of returning a callable, I was think we'd just return a list of mock blobs. The list should be able to be used downstream as a mock side effect too. If we want to keep this callable, we should rename the fixture `get_mock_blob_client` to make it a little more clear. I'm fine with either approach you prefer.